### PR TITLE
Feature/load sertificate from memory

### PR DESCRIPTION
**Pull request by VovaViter**

### CODE REVIEW

File: src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java [hunk 1]
```diff
@@ -21,12 +21,15 @@
...
 import java.security.cert.CertificateFactory;
 import java.security.cert.X509Certificate;
+import java.security.KeyFactory;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
 
 import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
+import org.bouncycastle.util.encoders.Base64;
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java:31]: @VovaViter 
do we really need Base64 from `bouncycastle` package? why do not use `java.util.Base64`?

File: src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java [hunk 2]
```diff
@@ -104,15 +108,17 @@ private static PublicKey getKeyFromConfigServer(RestTemplate keyUriRestTemplate)
...
-        log.info("Keystore location {}", applicationProperties.getKeystoreFile());
-        try (InputStream stream = new ClassPathResource(applicationProperties.getKeystoreFile()).exists()
-            ? new ClassPathResource(applicationProperties.getKeystoreFile()).getInputStream()
-            : new FileInputStream(new File(applicationProperties.getKeystoreFile()))) {
-            KeyStore store = KeyStore.getInstance(Constants.KEYSTORE_TYPE);
-            store.load(stream, applicationProperties.getKeystorePassword().toCharArray());
-            return (PrivateKey) store.getKey("selfsigned",
-                                             applicationProperties.getKeystorePassword().toCharArray());
+        NoSuchAlgorithmException, UnrecoverableKeyException, InvalidKeySpecException {
+        if (!org.springframework.util.StringUtils.isEmpty(applicationProperties.getPrivateKey())) {
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java:112]: please add an import for StringUtils

File: src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java [hunk 3]
```diff
@@ -104,15 +108,17 @@ private static PublicKey getKeyFromConfigServer(RestTemplate keyUriRestTemplate)
...
+            log.info("Keystore loaded from memory");
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java:113]: suggest changing log message to : `Key was loaded from memory by application property: app_property_name`

File: src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java [hunk 4]
```diff
@@ -124,4 +130,18 @@ public JwtTokenStore tokenStore(JwtAccessTokenConverter jwtAccessTokenConverter)
         return new JwtTokenStore(jwtAccessTokenConverter);
     }
 
+    private PrivateKey initPrivateKeyFromKeystore(InputStream stream) throws KeyStoreException, UnrecoverableKeyException,
+        NoSuchAlgorithmException, IOException, CertificateException {
+        KeyStore store = KeyStore.getInstance(Constants.KEYSTORE_TYPE);
+        store.load(stream, applicationProperties.getKeystorePassword().toCharArray());
+        return (PrivateKey) store.getKey("selfsigned",
```
Comment (amedvedchuk) [inline src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java:137]: I think it worth to extract key name inside keystore to properties with `swlfsigned` as default value

Comment (amedvedchuk) [reply src/main/java/com/icthh/xm/uaa/config/UaaAccessTokenConverterConfiguration.java:137]: parameter name like `kaystore-key-name`

**Review by amedvedchuk — APPROVED** [review]

**Review by sergeysenja1992 — APPROVED** [review]

**Pull request merged**